geode-docs/reference/topics/client-cache.html.md.erb

```diff
@@ -338,6 +338,32 @@ Provide a server list or `locator` list, but not both.
        port="123456"/>
 </pool>
 ```
+## <a id="cc-socket-factory" class="no-quick-link"></a>&lt;socket-factory&gt;
+
+Defines a factory to create socket connections to locators and servers.  A typical use of this element is to redirect connections to an ingress gateway such as Istio or HAProxy in a cluster where the TLS (SSL) Server Name Indication (SNI)field is set to indicate the actual locator or server the client is trying to reach.  This allows you to expose only the gateway hostname:port without the client needing to be able to resolve the names of the locator and server machines.
+
+**Note:**
+This setting may be used with either a Server list or a Locator list.  It will be used to form connections to either.
+
+**Default:**
+
+**API:** `org.apache.geode.cache.client.proxy.ProxySocketFactories`
+
+**Example:**
+
+``` pre
+<pool ...>
+ <socket-factory>
+    <class-name>org.apache.geode.cache.client.proxy.SniSocketFactory</class-name>
+    <parameter name="hostname">
+      <string>my-haproxy-address</string>
```

Comment:
for consistency, since the documentation refers to a "gateway" (not a "proxy") and also since HAproxy is only one of many choices, I recomend calling this something like `my-sni-gateway-address`

Reply:
done - it's now my-gateway-address like the other place